mewarpx/mewarpx/emission.py

```diff
@@ -236,26 +244,21 @@ def inject_particles(self):
             )
 
             print(f"Inject {len(particles_dict['x'])} particles")
-            # We can basically call _libwarpx.libwarpx.warpx_addNParticles()
-            # directly as we shouldn't need the wrapper, unless number to
-            # inject is 0, but for now we'll be conservative and use the
-            # wrapper.
 
             # Note some parts of WarpX call the variables ux and some parts vx,
             # and they're referred to as momenta. But I don't see anywhere
             # they're actually used as momenta including the particle mass -
             # the actual update is in Source/Particles/Pusher/UpdatePosition.H
-            _libwarpx.add_particles(
-                species_number=self.species.species_number,
+            self.species.add_particles(
                 x=particles_dict['x'],
                 y=particles_dict['y'],
                 z=particles_dict['z'],
                 ux=particles_dict['vx'],
                 uy=particles_dict['vy'],
                 uz=particles_dict['vz'],
-                # E_total=particles_dict['E_total'],
-                attr=particles_dict['w'],
-                unique_particles=self.unique_particles
+                w=particles_dict['w'],
+                unique_particles=self.unique_particles,
+                # E_total=particles_dict['E_total']
```

Comment:
Why is E_total commented out here?

Reply:
Yeah, let's have both injector types init and use E_total unless there's a clear reason not to.